### PR TITLE
Profiling: Support binding pprof server to custom network interfaces

### PR DESCRIPTION
**Pull request by cinaglia**
**What this PR does / why we need it**:

This PR tweaks the pprof server so it can listen on custom network interfaces (other than localhost). This is particularly useful when used in conjunction with a tool like [conprof](https://github.com/conprof/conprof).

**Which issue(s) this PR fixes**:

Listening on `0.0.0.0` allows us to access the `/debug/pprof*` endpoints without having to to rely on port-forwarding.

### CODE REVIEW

**Review by chaudyg — DISMISSED** [review]

**Comment by cinaglia** [comment]
> Might not be safe for everyone?

That's a valid concern. I'll tweak things so the default remains `localhost` and a custom address can be provided via `-profile-addr` or `GF_DIAGNOSTICS_PROFILING_ADDR`.

**Review by dafydd-t — APPROVED** [review]
I think will do what we need from a Hosted Grafana perspective.

File: docs/sources/troubleshooting/diagnostics.md [hunk 1]
```diff
@@ -10,11 +10,11 @@ when investigating certain performance problems. It's *not* recommended to have
 
 ## Turn on profiling
 
-The `grafana-server` can be started with the arguments `-profile` to enable profiling and  `-profile-port` to override
-the default HTTP port (`6060`) where the `pprof` debugging endpoints are available, for example:
+The `grafana-server` can be started with the arguments `-profile` to enable profiling, `-profile-addr` to override the default HTTP address (`localhost`), and
+`-profile-port` to override the default HTTP port (`6060`) where the `pprof` debugging endpoints are available, for example:
```
Comment (achatterjee-grafana) [inline docs/sources/troubleshooting/diagnostics.md:14]:
```suggestion
`-profile-port` to override the default HTTP port (`6060`) where the `pprof` debugging endpoints are available. For example:
```

File: docs/sources/troubleshooting/diagnostics.md [hunk 2]
```diff
@@ -52,4 +53,4 @@ go tool trace <trace file>
 2019/11/24 22:20:42 Opening browser. Trace viewer is listening on http://127.0.0.1:39735
 ```
 
-See [Go command trace](https://golang.org/cmd/trace/) for more information about how to analyze trace files.
\ No newline at end of file
+See [Go command trace](https://golang.org/cmd/trace/) for more information about how to analyze trace files.
```
Comment (achatterjee-grafana) [inline docs/sources/troubleshooting/diagnostics.md:56]:
```suggestion
For more information about how to analyze trace files, refer to [Go command trace](https://golang.org/cmd/trace/).
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added review comments.

File: docs/sources/troubleshooting/diagnostics.md (hunk 2 above)
Comment (cinaglia) [reply docs/sources/troubleshooting/diagnostics.md]: I didn't really change this sentence but I like the suggestion 👍

**Review by achatterjee-grafana — APPROVED** [review]

**Review by hairyhenderson — APPROVED** [review]

**Review by bergquist — APPROVED** [review]

**Comment by cinaglia** [comment]
Thanks all for the reviews. What's the process for getting this merged?

cc @bergquist

**Pull request merged**